crates/bindings-csharp/BSATN.Runtime/Builtins.cs

```diff
@@ -35,10 +160,50 @@ public readonly record struct Address
 
     internal Address(U128 v) => value = v;
 
+    /// <summary>
+    /// Create an Address from a LITTLE-ENDIAN byte array.
+    ///
+    /// If you are parsing an Address from a string, you probably want FromHexString instead,
+    /// or, failing that, FromBigEndian.
+    ///
+    /// Returns null if the resulting address is the default.
+    /// </summary>
+    /// <param name="bytes"></param>
     public static Address? From(byte[] bytes)
     {
         Debug.Assert(bytes.Length == 16);
-        var addr = new Address(MemoryMarshal.Read<U128>(bytes));
+        var addr = new Address(Util.ReadLittleEndian<U128>(bytes));
+        return addr == default ? null : addr;
+    }
+
+    /// <summary>
+    /// Create an Address from a BIG-ENDIAN byte array.
+    ///
+    /// This method is the correct choice if you have converted the bytes of a hexadecimal-formatted Address
+    /// to a byte array in the following way:
+    ///
+    /// "0xb0b1b2..."
+    /// ->
+    /// [0xb0, 0xb1, 0xb2, ...]
+    ///
+    /// Returns null if the resulting address is the default.
+    /// </summary>
+    /// <param name="bytes"></param>
+    public static Address? FromBigEndian(byte[] bytes)
+    {
+        Debug.Assert(bytes.Length == 16);
+        var addr = new Address(Util.ReadBigEndian<U128>(bytes));
+        return addr == default ? null : addr;
+    }
+
+    /// <summary>
+    /// Create an Address from a hex string.
+    /// </summary>
+    /// <param name="hex"></param>
+    /// <returns></returns>
+    public static Address? FromHexString(string hex)
+    {
+        var addr = new Address(Util.ReadFromBigEndianHexString<U128>(hex));
```

Comment:
Since this doesn't go via FromBigEndian, it should have its own length check. It probably needs its own anyway, since hex string parsing doesn't check for odd length.